src/main/java/au/org/emii/ncdfgenerator/cql/ExprParser.java

```diff
@@ -479,17 +482,24 @@ private ExprFloatLiteral parseFloatLiteral(String s, int pos) {
 
 	private ExprStringLiteral parseStringLiteral( String s, int pos )
 	{
-		// TODO use peekChar() not charAt()
-		// TODO ignore escaping for the moment
 		int pos2 = pos;
-		if(s.charAt(pos2) != '\'')
+		if (peekChar(s, pos2) != '\'') {
 			return null;
-		++pos2;
-
-		while(s.charAt(pos2) != '\'')
+		}
+		else {
 			++pos2;
-		++pos2;
-		return new ExprStringLiteral(pos, s.substring( pos + 1, pos2 - 1));
+
+			while (peekChar(s, pos2) != -1) {
+				++pos2;
+			}
```

Comment:
@anguss00 not sure if this loop looks right. It will glob to the end of the expression string. but we only want to get to the ' or " that closes the string literal. The original code should work, if the charAt() is replaced with peekChar().